src/content/blog/2025-11-aria-live.md

```diff
@@ -0,0 +1,428 @@
+setTimeout(() => (region.textContent = message), 50);
+```
+
+We will see that we don't have to do this by hand since there are very popular solutions for our frameworks already implementing this approach.
+
+✅ **Pros**
+
+- Always present in the DOM and therefore extremely reliable
+- Works across routes and components
+- Centralized and easy to test
+- Handles `polite` vs. `assertive` globally
+
+⚠️ **Cons**
+
+- Announcements lose some *local context* ("Where did that message come from?")
+- Requires a global setup or shared service
+
+## Implementing Reliable Live Regions in Angular, Vue, and React
+
+Now let's see how to make them work in practice — using the **global live announcer pattern**, since it's the most robust option across all three frameworks.
+
+### Angular
+
+Angular already ships an accessibility helper called [LiveAnnouncer](https://material.angular.dev/cdk/a11y/overview#liveannouncer) in the Angular CDK.
+
+```bash
+ng add @angular/cdk
+```
+
+Once you have installed the CDK (which I recommend since it also has other nice helpers for supporting accessibility), you can use the LiveAnnouncer as follows:
+
+```ts
+// save-button.ts
+import { LiveAnnouncer } from '@angular/cdk/a11y';
+import { Component, inject } from '@angular/core';
+
+@Component({
+  selector: 'app-save-button',
+  template: `<button (click)="onSave()">Save</button>`
+})
+export class SaveButton {
+  #liveAnnouncer = inject(LiveAnnouncer);
+
+  onSave() {
+    this.#liveAnnouncer.announce('Settings saved successfully.', 'polite');
+  }
+}
+```
+
+The CDK automatically creates a hidden live region and manages timing — no manual DOM work needed.
+
+### Vue 3
+
+For Vue applications, I recommend using [vue-a11y/vue-announcer](https://github.com/vue-a11y/vue-announcer).
+
+```bash
+npm install @vue-a11y/announcer@next # Vue 3
+# OR:
+npm install @vue-a11y/announcer      # Vue 2
+```
+
+Once installed, setup the `VueAnnouncer` for your `App`.
+
+```ts
+// main.ts
+import { createApp } from 'vue'
+import App from './App.vue'
+import router from './router'
+
+import VueAnnouncer from '@vue-a11y/announcer'
+import '@vue-a11y/announcer/dist/style.css'
+
+createApp(App)
+  .use(VueAnnouncer)
+  .use(router)
+  .mount('#app');
+```
+
+After that, place the component containing the global live region(s) in your main component:
+
+```html
+<!-- App.vue -->
+<template>
+  <VueAnnouncer class="sr-only" />
+  ...
+</template>
+```
+
+The last step is to use the composable `useAnnouncer` which pushes messages into the live region:
+
+```html
+<!-- SaveButton.vue -->
+<template>
+  <button @click="onSave">Save</button>
+</template>
+
+<script setup>
+import { useAnnouncer } from '@vue-a11y/announcer'
+
+const { polite } = useAnnouncer()
+
+function onSave() {
+  polite('Settings saved successfully.')
+}
+</script>
+```
+
+### React
+
+For React, I recommend using [@react-aria/live-announcer](https://react-spectrum.adobe.com/react-aria/useAnnouncer.html):
```

Comment:
Der Link geht ins Leere und ich kann in den aktuellen Docs von React Spektrum auch kein useAnnouncer mehr finden 😮